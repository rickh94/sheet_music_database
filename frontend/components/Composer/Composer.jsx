import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { withAlert } from 'react-alert'
import { connect } from 'react-redux'

import Container from 'react-bulma-components/lib/components/container'
import Box from 'react-bulma-components/lib/components/box'
import Level from 'react-bulma-components/lib/components/level'
import Heading from 'react-bulma-components/lib/components/heading'

import { composers } from '../../actions'
import { getDataOrLogIn } from '../../helpers'
import Header from '../Header'
import FieldDisplay from '../FieldDisplay'

export class Composer extends Component {
  static propTypes = {
    composers: PropTypes.object.isRequired,
    getComposer: PropTypes.func.isRequired,
    updateComposer: PropTypes.func.isRequired,
    alert: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired
  }

  state = {
    errors: {}
  }

  componentDidMount() {
    getDataOrLogIn(
      this.props.token,
      this.props.alert,
      this.props.history,
      this.props.getComposer,
      'Composer',
      [this.props.match.params.id]
    )
  }

  render() {
    const { name, born, died, era, short_name } = this.props.composers.composer
    return (
      <React.Fragment>
        <Header />
        <Container>
          <Box>
            <Level>
              <Heading size={3} className="absolutely-no-margin level-left">
                <FieldDisplay
                  value={name}
                  saveCallback={() => {}}
                  backendFieldName="name"
                  errors={this.state.errors.name}
                />
              </Heading>
            </Level>
            <FieldDisplay
              value={era}
              saveCallback={() => {}}
              backendFieldName="era"
              label="Era"
              errors={this.state.errors.era}
            />
            <FieldDisplay
              value={short_name}
              saveCallback={() => {}}
              backendFieldName="short_name"
              label="Short Name"
              errors={this.state.errors.short_name}
            />
          </Box>
        </Container>
      </React.Fragment>
    )
  }
}

const mapStateToProps = state => {
  return {
    token: state.account.token,
    composers: state.composers
  }
}

const mapDispatchToProps = dispatch => {
  return {
    updateComposer: (token, id, updated) => {
      return dispatch(composers.updateSingleComposer(token, id, updated))
    },
    getComposer: (token, id) => {
      return dispatch(composers.getComposer(token, id))
    }
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withAlert(Composer))
