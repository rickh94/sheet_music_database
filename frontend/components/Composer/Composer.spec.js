import React from 'react'
import { shallow } from 'enzyme'
import '../../setupTests'

import {
  testToken,
  testLoggedIn,
  testGetData,
  testGetDataFailed
} from '../../testHelpers'

import { Composer } from './Composer'

describe('Composer', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <Composer
        alert={{ show: jest.fn() }}
        getComposer={jest.fn()}
        updateComposer={jest.fn()}
        token={testToken}
        match={{ params: { id: 1 } }}
        composers={{ composer: {} }}
        history={{ push: jest.fn() }}
      />
    )
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the header', () => {
    expect(wrapper.exists('Connect(Header)')).toBeTruthy()
  })

  it('renders a container', () => {
    expect(wrapper.exists('Container')).toBeTruthy()
  })

  it('renders a composer from props', () => {
    const composer = {
      id: 1,
      name: 'Test Name',
      born: '2018-01-02',
      died: '2018-01-03',
      era: 'test era',
      short_name: 'test'
    }
    wrapper.setProps({ composer })
    expect(wrapper.find('FieldDisplay').length).toEqual(5)
  })

  it('redirects if not logged in', () => {
    testLoggedIn(wrapper)
  })

  it('gets the composer', () => {
    testGetData(wrapper, 'getComposer', [1])
  })

  it('shows error on get failure', () => {
    testGetDataFailed(wrapper, 'getComposer', 'Composer')
  })

  it('calls updateComposer with updated data', () => {
    const updateComposer = jest.fn()
    wrapper.setProps({ token: 'testtoken', updateComposer })
    wrapper.instance().updateField('test', 'newValue')
    expect(updateComposer).toHaveBeenCalledWith('testtoken', 1, { test: 'newValue' })
  })

  it('sets errors in state on error', async () => {
    const updateComposer = (token, id, updated) => false
    const errors = { name: 'invalid name' }
    wrapper.setProps({ updateComposer, composers: { errors, composer: {} } })
    await wrapper.instance().updateField('test', 'test')
    expect(wrapper.state().errors).toEqual(errors)
  })
})
