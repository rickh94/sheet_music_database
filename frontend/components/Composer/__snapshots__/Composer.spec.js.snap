// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Composer matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Composer
    alert={
      Object {
        "show": [MockFunction] {
          "calls": Array [
            Array [
              <span
                className="alert-text"
              >
                Could not get Composer
              </span>,
              Object {
                "type": "error",
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
      }
    }
    composers={
      Object {
        "composer": Object {},
      }
    }
    getComposer={
      [MockFunction] {
        "calls": Array [
          Array [
            "testtoken",
            1,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
    token="testtoken"
    updateComposer={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <Container
          breakpoint={null}
          fluid={false}
          renderAs="div"
          style={Object {}}
        >
          <Box
            className=""
            renderAs="div"
            style={Object {}}
          >
            <Level
              breakpoint={null}
              className=""
              renderAs="div"
              style={Object {}}
            >
              <Heading
                className="absolutely-no-margin level-left"
                heading={false}
                renderAs="h1"
                size={3}
                spaced={false}
                subtitle={false}
                weight={null}
              >
                <FieldDisplay
                  backendFieldName="name"
                  saveCallback={[Function]}
                  value=""
                />
              </Heading>
            </Level>
            <FieldDisplay
              backendFieldName="era"
              label="Era"
              saveCallback={[Function]}
              value=""
            />
            <FieldDisplay
              backendFieldName="short_name"
              label="Short Name"
              saveCallback={[Function]}
              value=""
            />
            <FieldDisplay
              backendFieldName="born"
              label="Born"
              saveCallback={[Function]}
              type="date"
              value={null}
            />
            <FieldDisplay
              backendFieldName="died"
              label="Died"
              saveCallback={[Function]}
              type="date"
              value={null}
            />
          </Box>
        </Container>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "backgroundColor": undefined,
          "breakpoint": null,
          "children": <Box
            className=""
            renderAs="div"
            style={Object {}}
          >
            <Level
              breakpoint={null}
              className=""
              renderAs="div"
              style={Object {}}
            >
              <Heading
                className="absolutely-no-margin level-left"
                heading={false}
                renderAs="h1"
                size={3}
                spaced={false}
                subtitle={false}
                weight={null}
              >
                <FieldDisplay
                  backendFieldName="name"
                  saveCallback={[Function]}
                  value=""
                />
              </Heading>
            </Level>
            <FieldDisplay
              backendFieldName="era"
              label="Era"
              saveCallback={[Function]}
              value=""
            />
            <FieldDisplay
              backendFieldName="short_name"
              label="Short Name"
              saveCallback={[Function]}
              value=""
            />
            <FieldDisplay
              backendFieldName="born"
              label="Born"
              saveCallback={[Function]}
              type="date"
              value={null}
            />
            <FieldDisplay
              backendFieldName="died"
              label="Died"
              saveCallback={[Function]}
              type="date"
              value={null}
            />
          </Box>,
          "className": undefined,
          "clearfix": undefined,
          "clipped": undefined,
          "fluid": false,
          "hidden": undefined,
          "invisible": undefined,
          "italic": undefined,
          "marginless": undefined,
          "overlay": undefined,
          "paddingless": undefined,
          "pull": undefined,
          "radiusless": undefined,
          "renderAs": "div",
          "responsive": undefined,
          "shadowless": undefined,
          "style": Object {},
          "textAlignment": undefined,
          "textColor": undefined,
          "textSize": undefined,
          "textTransform": undefined,
          "textWeight": undefined,
          "unselectable": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backgroundColor": undefined,
            "children": Array [
              <Level
                breakpoint={null}
                className=""
                renderAs="div"
                style={Object {}}
              >
                <Heading
                  className="absolutely-no-margin level-left"
                  heading={false}
                  renderAs="h1"
                  size={3}
                  spaced={false}
                  subtitle={false}
                  weight={null}
                >
                  <FieldDisplay
                    backendFieldName="name"
                    saveCallback={[Function]}
                    value=""
                  />
                </Heading>
              </Level>,
              <FieldDisplay
                backendFieldName="era"
                label="Era"
                saveCallback={[Function]}
                value=""
              />,
              <FieldDisplay
                backendFieldName="short_name"
                label="Short Name"
                saveCallback={[Function]}
                value=""
              />,
              <FieldDisplay
                backendFieldName="born"
                label="Born"
                saveCallback={[Function]}
                type="date"
                value={null}
              />,
              <FieldDisplay
                backendFieldName="died"
                label="Died"
                saveCallback={[Function]}
                type="date"
                value={null}
              />,
            ],
            "className": "",
            "clearfix": undefined,
            "clipped": undefined,
            "hidden": undefined,
            "invisible": undefined,
            "italic": undefined,
            "marginless": undefined,
            "overlay": undefined,
            "paddingless": undefined,
            "pull": undefined,
            "radiusless": undefined,
            "renderAs": "div",
            "responsive": undefined,
            "shadowless": undefined,
            "style": Object {},
            "textAlignment": undefined,
            "textColor": undefined,
            "textSize": undefined,
            "textTransform": undefined,
            "textWeight": undefined,
            "unselectable": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "backgroundColor": undefined,
                "breakpoint": null,
                "children": <Heading
                  className="absolutely-no-margin level-left"
                  heading={false}
                  renderAs="h1"
                  size={3}
                  spaced={false}
                  subtitle={false}
                  weight={null}
                >
                  <FieldDisplay
                    backendFieldName="name"
                    saveCallback={[Function]}
                    value=""
                  />
                </Heading>,
                "className": "",
                "clearfix": undefined,
                "clipped": undefined,
                "hidden": undefined,
                "invisible": undefined,
                "italic": undefined,
                "marginless": undefined,
                "overlay": undefined,
                "paddingless": undefined,
                "pull": undefined,
                "radiusless": undefined,
                "renderAs": "div",
                "responsive": undefined,
                "shadowless": undefined,
                "style": Object {},
                "textAlignment": undefined,
                "textColor": undefined,
                "textSize": undefined,
                "textTransform": undefined,
                "textWeight": undefined,
                "unselectable": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "backgroundColor": undefined,
                  "children": <FieldDisplay
                    backendFieldName="name"
                    saveCallback={[Function]}
                    value=""
                  />,
                  "className": "absolutely-no-margin level-left",
                  "clearfix": undefined,
                  "clipped": undefined,
                  "heading": false,
                  "hidden": undefined,
                  "invisible": undefined,
                  "italic": undefined,
                  "marginless": undefined,
                  "overlay": undefined,
                  "paddingless": undefined,
                  "pull": undefined,
                  "radiusless": undefined,
                  "renderAs": "h1",
                  "responsive": undefined,
                  "shadowless": undefined,
                  "size": 3,
                  "spaced": false,
                  "subtitle": false,
                  "textAlignment": undefined,
                  "textColor": undefined,
                  "textSize": undefined,
                  "textTransform": undefined,
                  "textWeight": undefined,
                  "unselectable": undefined,
                  "weight": null,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "backendFieldName": "name",
                    "errors": undefined,
                    "saveCallback": [Function],
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "backendFieldName": "era",
                "errors": undefined,
                "label": "Era",
                "saveCallback": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "backendFieldName": "short_name",
                "errors": undefined,
                "label": "Short Name",
                "saveCallback": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "backendFieldName": "born",
                "errors": undefined,
                "label": "Born",
                "saveCallback": [Function],
                "type": "date",
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "backendFieldName": "died",
                "errors": undefined,
                "label": "Died",
                "saveCallback": [Function],
                "type": "date",
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <Container
            breakpoint={null}
            fluid={false}
            renderAs="div"
            style={Object {}}
          >
            <Box
              className=""
              renderAs="div"
              style={Object {}}
            >
              <Level
                breakpoint={null}
                className=""
                renderAs="div"
                style={Object {}}
              >
                <Heading
                  className="absolutely-no-margin level-left"
                  heading={false}
                  renderAs="h1"
                  size={3}
                  spaced={false}
                  subtitle={false}
                  weight={null}
                >
                  <FieldDisplay
                    backendFieldName="name"
                    saveCallback={[Function]}
                    value=""
                  />
                </Heading>
              </Level>
              <FieldDisplay
                backendFieldName="era"
                label="Era"
                saveCallback={[Function]}
                value=""
              />
              <FieldDisplay
                backendFieldName="short_name"
                label="Short Name"
                saveCallback={[Function]}
                value=""
              />
              <FieldDisplay
                backendFieldName="born"
                label="Born"
                saveCallback={[Function]}
                type="date"
                value={null}
              />
              <FieldDisplay
                backendFieldName="died"
                label="Died"
                saveCallback={[Function]}
                type="date"
                value={null}
              />
            </Box>
          </Container>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backgroundColor": undefined,
            "breakpoint": null,
            "children": <Box
              className=""
              renderAs="div"
              style={Object {}}
            >
              <Level
                breakpoint={null}
                className=""
                renderAs="div"
                style={Object {}}
              >
                <Heading
                  className="absolutely-no-margin level-left"
                  heading={false}
                  renderAs="h1"
                  size={3}
                  spaced={false}
                  subtitle={false}
                  weight={null}
                >
                  <FieldDisplay
                    backendFieldName="name"
                    saveCallback={[Function]}
                    value=""
                  />
                </Heading>
              </Level>
              <FieldDisplay
                backendFieldName="era"
                label="Era"
                saveCallback={[Function]}
                value=""
              />
              <FieldDisplay
                backendFieldName="short_name"
                label="Short Name"
                saveCallback={[Function]}
                value=""
              />
              <FieldDisplay
                backendFieldName="born"
                label="Born"
                saveCallback={[Function]}
                type="date"
                value={null}
              />
              <FieldDisplay
                backendFieldName="died"
                label="Died"
                saveCallback={[Function]}
                type="date"
                value={null}
              />
            </Box>,
            "className": undefined,
            "clearfix": undefined,
            "clipped": undefined,
            "fluid": false,
            "hidden": undefined,
            "invisible": undefined,
            "italic": undefined,
            "marginless": undefined,
            "overlay": undefined,
            "paddingless": undefined,
            "pull": undefined,
            "radiusless": undefined,
            "renderAs": "div",
            "responsive": undefined,
            "shadowless": undefined,
            "style": Object {},
            "textAlignment": undefined,
            "textColor": undefined,
            "textSize": undefined,
            "textTransform": undefined,
            "textWeight": undefined,
            "unselectable": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "backgroundColor": undefined,
              "children": Array [
                <Level
                  breakpoint={null}
                  className=""
                  renderAs="div"
                  style={Object {}}
                >
                  <Heading
                    className="absolutely-no-margin level-left"
                    heading={false}
                    renderAs="h1"
                    size={3}
                    spaced={false}
                    subtitle={false}
                    weight={null}
                  >
                    <FieldDisplay
                      backendFieldName="name"
                      saveCallback={[Function]}
                      value=""
                    />
                  </Heading>
                </Level>,
                <FieldDisplay
                  backendFieldName="era"
                  label="Era"
                  saveCallback={[Function]}
                  value=""
                />,
                <FieldDisplay
                  backendFieldName="short_name"
                  label="Short Name"
                  saveCallback={[Function]}
                  value=""
                />,
                <FieldDisplay
                  backendFieldName="born"
                  label="Born"
                  saveCallback={[Function]}
                  type="date"
                  value={null}
                />,
                <FieldDisplay
                  backendFieldName="died"
                  label="Died"
                  saveCallback={[Function]}
                  type="date"
                  value={null}
                />,
              ],
              "className": "",
              "clearfix": undefined,
              "clipped": undefined,
              "hidden": undefined,
              "invisible": undefined,
              "italic": undefined,
              "marginless": undefined,
              "overlay": undefined,
              "paddingless": undefined,
              "pull": undefined,
              "radiusless": undefined,
              "renderAs": "div",
              "responsive": undefined,
              "shadowless": undefined,
              "style": Object {},
              "textAlignment": undefined,
              "textColor": undefined,
              "textSize": undefined,
              "textTransform": undefined,
              "textWeight": undefined,
              "unselectable": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "backgroundColor": undefined,
                  "breakpoint": null,
                  "children": <Heading
                    className="absolutely-no-margin level-left"
                    heading={false}
                    renderAs="h1"
                    size={3}
                    spaced={false}
                    subtitle={false}
                    weight={null}
                  >
                    <FieldDisplay
                      backendFieldName="name"
                      saveCallback={[Function]}
                      value=""
                    />
                  </Heading>,
                  "className": "",
                  "clearfix": undefined,
                  "clipped": undefined,
                  "hidden": undefined,
                  "invisible": undefined,
                  "italic": undefined,
                  "marginless": undefined,
                  "overlay": undefined,
                  "paddingless": undefined,
                  "pull": undefined,
                  "radiusless": undefined,
                  "renderAs": "div",
                  "responsive": undefined,
                  "shadowless": undefined,
                  "style": Object {},
                  "textAlignment": undefined,
                  "textColor": undefined,
                  "textSize": undefined,
                  "textTransform": undefined,
                  "textWeight": undefined,
                  "unselectable": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "backgroundColor": undefined,
                    "children": <FieldDisplay
                      backendFieldName="name"
                      saveCallback={[Function]}
                      value=""
                    />,
                    "className": "absolutely-no-margin level-left",
                    "clearfix": undefined,
                    "clipped": undefined,
                    "heading": false,
                    "hidden": undefined,
                    "invisible": undefined,
                    "italic": undefined,
                    "marginless": undefined,
                    "overlay": undefined,
                    "paddingless": undefined,
                    "pull": undefined,
                    "radiusless": undefined,
                    "renderAs": "h1",
                    "responsive": undefined,
                    "shadowless": undefined,
                    "size": 3,
                    "spaced": false,
                    "subtitle": false,
                    "textAlignment": undefined,
                    "textColor": undefined,
                    "textSize": undefined,
                    "textTransform": undefined,
                    "textWeight": undefined,
                    "unselectable": undefined,
                    "weight": null,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "backendFieldName": "name",
                      "errors": undefined,
                      "saveCallback": [Function],
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "backendFieldName": "era",
                  "errors": undefined,
                  "label": "Era",
                  "saveCallback": [Function],
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "backendFieldName": "short_name",
                  "errors": undefined,
                  "label": "Short Name",
                  "saveCallback": [Function],
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "backendFieldName": "born",
                  "errors": undefined,
                  "label": "Born",
                  "saveCallback": [Function],
                  "type": "date",
                  "value": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "backendFieldName": "died",
                  "errors": undefined,
                  "label": "Died",
                  "saveCallback": [Function],
                  "type": "date",
                  "value": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
